### DISCUSSION TITLE
Ability to Change Message Composer

### DISCUSSION THREAD

**Discussion by arcxne** [d99kris/nchat#228] [Ideas]
I usually use the message compose function to write my messages instead of typing directly (due to my keybindings). An annoyance I face is that after writing my message, I have to press ctrl-x, y, enter just to save my message, whereas in vim, this would just be esc, :wq.

As such, would it be possible to allow users to change the default message compose to their preferred editor (e.g. vim, emacs)? This could be added inside app.conf and users can change the default similar to how the message viewer can be changed.

**Comment by d99kris** [comment]
Hi @arcxne - agree it makes sense to add a config option for this, will look into it. 

As a workaround you can set the `EDITOR` environment variable, either `export EDITOR=vim` in `~/.bashrc` or just for the nchat process, example: `EDITOR=vim nchat`

**Reply by d99kris** [reply]
A configuration parameter `message_edit_command` has been added now in `ui.conf` (available in nchat v4.80 onwards). 

To use `vim` one can set it like this: `message_edit_command=vim`
